.cz.json

```diff
@@ -0,0 +1,48 @@
+{
+  "commitizen": {
+    "name": "cz_customize",
+    "version_scheme": "semver",
+    "version_provider": "npm",
+    "update_changelog_on_bump": true,
+    "major_version_zero": true,
+    "bump_message": "chore: release $new_version",
+    "gpg_sign": true,
+    "changelog_incremental": true,
+    "customize": {
+      "message_template": "{{change_type}}:{% if show_message %} {{message}}{% endif %}",
+      "example": "feat: this feature enable customize through config file",
+      "schema": "<type>: <body>",
+      "schema_pattern": "^((build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test){1}(\\([\\w\\-\\.]+\\))?(!)?: ([\\w \\-'])+([\\s\\S]*))|Merge.*",
+      "bump_pattern": "^(break|hotfix|chore|docs|feat|fix|perf|refactor|revert|style|test)",
+      "bump_map": {
+          "break": "MAJOR",
+          "hotfix": "PATCH",
+          "feat": "MINOR",
+          "fix": "PATCH",
+          "chore": "PATCH",
+          "docs": "PATCH",
+          "perf": "PATCH",
+          "refactor": "PATCH",
+          "revert": "MINOR",
+          "style": "PATCH",
+          "test": "PATCH"
+      },
+      "change_type_order": ["BREAKING CHANGE", "feat", "fix", "perf", "docs", "revert", "chore", "refactor", "style", "test"],
+      "info_path": "cz_customize_info.txt",
+      "info": "This is customized info",
+      "commit_parser": "^(?P<change_type>chore|docs|feat|fix|perf|refactor|revert|style|test):\\s(?P<message>.*)?",
+      "changelog_pattern": "^(chore|docs|feat|fix|perf|refactor|revert|style|test)?(!)?",
+      "change_type_map": {
```

Comment:
Should the `change_type_map` and the other change related key values match the `bump_map` values? Is there a situation where want to use the `hotfix` keyword to affect a bump but not the changelog?

Reply:
This is a good question, I think these differences probably aren't necessary and we should match our schema pattern almost exactly (leaving out `build` to trigger builds without changing version). Breaking changes can be indicated with `!` after the type and hotfixes can be tagged normally (e.g. `fix`).

Reply:
Actually this isn't exactly true, conventional commits spec says we should major bump on either `!` or `BREAKING CHANGE` in the footer and I'll need to add a custom rule for that, working on it now

Reply:
The updated version should correctly handle breaking changes, indicated either via the footer or with `!` with one small exception: if someone uses the `feat!` notation with no footer, it will end up in the `Added` section rather than `Breaking Changes` but I can live with that for now

Reply:
Sounds good.